### PR TITLE
Split out mods category sections

### PR DESCRIPTION
**Pull request by peppy**
- Splits out "Special" into "Automation" and "Conversion".
- Makes mod select scrollable (with hidden columns).
- Adds new (empty) "Fun" section.
- Tidies some UI code up.

### CODE REVIEW

File: osu.Game.Rulesets.Osu/OsuRuleset.cs [hunk 1]
```diff
@@ -94,6 +94,7 @@ public override IEnumerable<Mod> GetModsFor(ModType type)
                         new OsuModEasy(),
                         new OsuModNoFail(),
                         new MultiMod(new OsuModHalfTime(), new OsuModDaycore()),
+                        new OsuModSpunOut(),
```
Comment (Aergwyn) [inline osu.Game.Rulesets.Osu/OsuRuleset.cs]: Couldn't this be regarded as automation as well?

File: osu.Game/Rulesets/UI/ModIcon.cs [hunk 2]
```diff
@@ -92,10 +100,7 @@ private void load(OsuColour colours)
 
         public bool Highlighted
         {
-            get
-            {
-                return highlighted;
-            }
+            get { return highlighted; }
```
Comment (Aergwyn) [inline osu.Game/Rulesets/UI/ModIcon.cs]: `get => highlighted;`

File: osu.Game.Rulesets.Osu/OsuRuleset.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Rulesets.Osu/OsuRuleset.cs]: Moved intentionally as it is ranked and has a negative multiplier. All remaining automation mods are non-ranked, which is better defined IMO.

File: osu.Game/Rulesets/UI/ModIcon.cs (hunk 2 above)
Comment (peppy) [reply osu.Game/Rulesets/UI/ModIcon.cs]: eh

**Review by smoogipoo — APPROVED** [review]
Seems fine to me.

**Pull request merged**